atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -85,19 +86,26 @@ public static OracleDdlTable create(
 
     @Override
     public void create(byte[] tableMetadata) {
+        boolean needsOverflow = needsOverflow(tableMetadata);
+
         if (conns.get().selectExistsUnregisteredQuery(
                 "SELECT 1 FROM " + config.metadataTable().getQualifiedName() + " WHERE table_name = ?",
                 tableRef.getQualifiedName())) {
+            if (needsOverflow) {
+                TableValueStyle existingStyle = valueStyleCache.getTableType(conns, tableRef, config.metadataTable());
+                if (existingStyle != TableValueStyle.OVERFLOW) {
+                    throwForMissingOverflowTable();
+                }
+            }
             return;
         }
 
-        boolean needsOverflow = false;
-        TableMetadata metadata = TableMetadata.BYTES_HYDRATOR.hydrateFromBytes(tableMetadata);
-        if (metadata != null) {
-            needsOverflow = metadata.getColumns().getMaxValueSize() > AtlasDbConstants.ORACLE_OVERFLOW_THRESHOLD;
+        String shortTableName = createTable(needsOverflow);
+
+        if (needsOverflow && !overflowColumnExists(shortTableName)) {
+            throwForMissingOverflowTable();
```

Comment:
This part is to handle races between multiple nodes in an upgrade right?

Reply:
It could handle races or simply startup failures. `createTable` is the first modification we make, and the last one is to insert the table into the metadata table, so any failure in between would leave us in an inconsistent state that wouldn't be caught by the first `throwForMissingOverflowTable`.